src/hammer-vlsi/technology/asap7/__init__.py

```diff
@@ -244,13 +244,16 @@ def asap7_scale_final_gds(ht: HammerTool) -> bool:
     Scale the final GDS by a factor of 4
     scale_gds_script writes the actual Python script to execute from the Tcl interpreter
     """
+    # This is from the Cadence tools
+    cadence_output_gds_name = ht.output_gds_filename  # type: ignore
```

Comment:
Ugh, this isn't elegant. Any proposal for how not to have to have these extra lines?

Reply:
Hmm. One possible resolution is to add `output_gds_filename` into the core Hammer P&R abstraction. Currently this is actually problematic from a type-checker perspective, since it has no idea of knowing that `output_gds_filename` is valid because that is defined in CadenceTool, not core Hammer.